server/src/main/java/org/opensearch/search/aggregations/metrics/AvgAggregator.java

```diff
@@ -226,4 +233,40 @@ public void doClose() {
         Releasables.close(counts, sums, compensations);
     }
 
+    public StarTreeBucketCollector getStarTreeBucketCollector(
+        LeafReaderContext ctx,
+        CompositeIndexFieldInfo starTree,
+        StarTreeBucketCollector parentCollector
+    ) throws IOException {
+        assert parentCollector != null;
+        return new StarTreeBucketCollector(parentCollector) {
+            String sumMetricName = StarTreeUtils.fullyQualifiedFieldNameForStarTreeMetricsDocValues(
+                starTree.getField(),
+                ((ValuesSource.Numeric.FieldData) valuesSource).getIndexFieldName(),
+                MetricStat.SUM.getTypeName()
+            );
+            String valueCountMetricName = StarTreeUtils.fullyQualifiedFieldNameForStarTreeMetricsDocValues(
+                starTree.getField(),
+                ((ValuesSource.Numeric.FieldData) valuesSource).getIndexFieldName(),
+                MetricStat.VALUE_COUNT.getTypeName()
+            );
+            SortedNumericStarTreeValuesIterator sumMetricValuesIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+                .getMetricValuesIterator(sumMetricName);
+            SortedNumericStarTreeValuesIterator valueCountMetricValuesIterator = (SortedNumericStarTreeValuesIterator) starTreeValues
+                .getMetricValuesIterator(valueCountMetricName);
+
+            @Override
+            public void collectStarTreeEntry(int starTreeEntryBit, long bucket) throws IOException {
+                counts = context.bigArrays().grow(counts, bucket + 1);
+                sums = context.bigArrays().grow(sums, bucket + 1);
+                // Advance the valuesIterator to the current bit
+                if (!sumMetricValuesIterator.advanceExact(starTreeEntryBit)
+                    || !valueCountMetricValuesIterator.advanceExact(starTreeEntryBit)) {
+                    return; // Skip if no entries for this document
+                }
+                counts.increment(bucket, valueCountMetricValuesIterator.nextValue());
+                sums.set(bucket, NumericUtils.sortableLongToDouble(sumMetricValuesIterator.nextValue()) + sums.get(bucket));
```

Comment:
Lets use kahan summation similar to the older star tree collector and default collector - also lets add some tests where without kahan summation there will be errors to assert precision if possible, tests are not a blocker for 2.19

Reply:
Gotcha - making the changes - might have to capture the tests in a follow-up PR specifically for this.